src/activation.js

```diff
@@ -180,11 +180,58 @@ function shouldContinue(output, router: Router) {
   return output;
 }
 
+//wraps a subscription, allowing unsubscribe calls even if
+//the first value comes synchronously
+class SafeSubscription {
+  //if this were TypeScript, subscriptionFunc would be of type
+  //(sub: SafeSubscription) => Subscription
```

Comment:
Yeah, they're pretty similar. Gulp will spit out a d.ts file that you can review if you want to confirm.